fenics_ice/model.py

```diff
@@ -362,6 +374,31 @@ def interpolate(values, vtx, wts):
         # it is the mask from the composite velocities
         self.mask_vel_M.vector()[:] = interpolate(self.vel_obs['mask_vel'], vtx_M, wts_M)
 
+        # We need to do the same as above but for cloud point data
+        # so we can write out a nicer output in the mesh coordinates
+        vtx_Q_c, wts_Q_c = interp_weights(self.vel_obs['uv_obs_pts'],
+                                      Q_coords)
+        vtx_M_c, wts_M_c = interp_weights(self.vel_obs['uv_obs_pts'],
+                                      M_coords)
+
+        # Define new functions to hold results
+        self.u_cloud_Q = Function(self.Q, name="u_obs_cloud", static=True)
+        self.v_cloud_Q = Function(self.Q, name="v_obs_cloud", static=True)
+        self.u_std_cloud_Q = Function(self.Q, name="u_std_cloud", static=True)
+        self.v_std_cloud_Q = Function(self.Q, name="v_std_cloud", static=True)
+        # There is no mask for this data!
+        self.u_cloud_M = Function(self.M, name="u_obs_cloud", static=True)
+        self.v_cloud_M = Function(self.M, name="v_obs_cloud", static=True)
+
+        # Fill via interpolation
+        self.u_cloud_Q.vector()[:] = interpolate(self.vel_obs['u_obs'], vtx_Q_c, wts_Q_c)
+        self.v_cloud_Q.vector()[:] = interpolate(self.vel_obs['v_obs'], vtx_Q_c, wts_Q_c)
+        self.u_std_cloud_Q.vector()[:] = interpolate(self.vel_obs['u_std'], vtx_Q_c, wts_Q_c)
+        self.v_std_cloud_Q.vector()[:] = interpolate(self.vel_obs['v_std'], vtx_Q_c, wts_Q_c)
+
+        self.u_cloud_M.vector()[:] = interpolate(self.vel_obs['u_obs'], vtx_M_c, wts_M_c)
+        self.v_cloud_M.vector()[:] = interpolate(self.vel_obs['v_obs'], vtx_M_c, wts_M_c)
```

Comment:
Should different fields be interpolated here? Are `u_cloud_M` and `v_cloud_M` needed?

Reply:
good point only `u_cloud_Q` gets written I will delete this